hazelcast/src/main/java/com/hazelcast/internal/monitors/BuildInfoPlugin.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.monitors;
+
+import com.hazelcast.instance.BuildInfo;
+import com.hazelcast.instance.BuildInfoProvider;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.spi.impl.NodeEngineImpl;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+
+/**
+ * A {@link PerformanceMonitorPlugin} that displays the build info.
+ */
+public class BuildInfoPlugin extends PerformanceMonitorPlugin {
+
+    private final BuildInfo buildInfo = BuildInfoProvider.getBuildInfo();
+    private final ILogger logger;
+
+    public BuildInfoPlugin(NodeEngineImpl nodeEngine) {
+        this.logger = nodeEngine.getLogger(BuildInfoPlugin.class);
+    }
+
+    @Override
+    public long getPeriodMillis() {
+        return -1;
```

Comment:
This looks like a magic number. I think I'll see this further in the code, but if this means "execute once" it would be nice to have a constant in `PerformanceMonitorPlugin` for this special value.